src/app/header/header.component.ts

```diff
@@ -0,0 +1,19 @@
+import { Component } from '@angular/core';
+
+@Component({
+  selector: 'app-header',
+  templateUrl: './header.component.html',
+  styleUrls: ['./header.component.css']
+})
+export class HeaderComponent {
+  export() {
```

Comment:
please wrap your logic with service, inject that service on widget component and implement